# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/sandbox/linux/seccomp-bpf/sandbox_bpf.cc
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 3
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium/sandbox/linux/seccomp-bpf/sandbox_bpf.cc
--- a/security/sandbox/chromium/sandbox/linux/seccomp-bpf/sandbox_bpf.cc	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium/sandbox/linux/seccomp-bpf/sandbox_bpf.cc	Wed Jun 12 11:10:48 2019 +0100
@@ -15,7 +15,6 @@
 #include "base/logging.h"
 #include "base/macros.h"
 #include "base/posix/eintr_wrapper.h"
-#include "base/third_party/valgrind/valgrind.h"
 #include "sandbox/linux/bpf_dsl/bpf_dsl.h"
 #include "sandbox/linux/bpf_dsl/codegen.h"
 #include "sandbox/linux/bpf_dsl/policy.h"
@@ -36,8 +35,6 @@
 
 namespace {
 
-bool IsRunningOnValgrind() { return RUNNING_ON_VALGRIND; }
-
 // Check if the kernel supports seccomp-filter (a.k.a. seccomp mode 2) via
 // prctl().
 bool KernelSupportsSeccompBPF() {
@@ -90,12 +87,11 @@
 
   if (rv == -1 && errno == EFAULT) {
     return true;
-  } else {
-    // TODO(jln): turn these into DCHECK after 417888 is considered fixed.
-    CHECK_EQ(-1, rv);
-    CHECK(ENOSYS == errno || EINVAL == errno);
-    return false;
   }
+
+  DCHECK_EQ(-1, rv);
+  DCHECK(ENOSYS == errno || EINVAL == errno);
+  return false;
 }
 
 uint64_t EscapePC() {
@@ -116,21 +112,14 @@
 
 }  // namespace
 
-SandboxBPF::SandboxBPF(bpf_dsl::Policy* policy)
-    : proc_fd_(), sandbox_has_started_(false), policy_(policy) {
-}
+SandboxBPF::SandboxBPF(std::unique_ptr<bpf_dsl::Policy> policy)
+    : proc_fd_(), sandbox_has_started_(false), policy_(std::move(policy)) {}
 
 SandboxBPF::~SandboxBPF() {
 }
 
 // static
 bool SandboxBPF::SupportsSeccompSandbox(SeccompLevel level) {
-  // Never pretend to support seccomp with Valgrind, as it
-  // throws the tool off.
-  if (IsRunningOnValgrind()) {
-    return false;
-  }
-
   switch (level) {
     case SeccompLevel::SINGLE_THREADED:
       return KernelSupportsSeccompBPF();